seqr/utils/search/utils.py

```diff
@@ -96,37 +115,57 @@ def query_variants(search_model, es_search_cls=EsSearch, sort=XPOS_SORT_KEY, ski
         if rs_ids and variant_ids:
             raise InvalidSearchException('Invalid variant notation: found both variant IDs and rsIDs')
 
-    es_search = es_search_cls(
-        search_model.families.all(),
-        previous_search_results=previous_search_results,
-        user=user,
-        sort=sort,
-    )
-
-    es_search.filter_variants(
-        inheritance=search.get('inheritance'), frequencies=search.get('freqs'), pathogenicity=search.get('pathogenicity'),
-        annotations=search.get('annotations'), annotations_secondary=search.get('annotations_secondary'),
-        in_silico=search.get('in_silico'), quality_filter=search.get('qualityFilter'),
-        custom_query=search.get('customQuery'), locus=search.get('locus'),
-        genes=genes, intervals=intervals, rs_ids=rs_ids, variant_ids=variant_ids,
-        skip_genotype_filter=skip_genotype_filter,
-    )
-
     if variant_ids:
         num_results = len(variant_ids)
 
-    variant_results = es_search.search(page=page, num_results=num_results)
+    parsed_search = {
+        'parsedLocus': {
+            'genes': genes, 'intervals': intervals, 'rs_ids': rs_ids, 'variant_ids': variant_ids,
+        },
+    }
+    parsed_search.update(search)
 
-    safe_redis_set_json(cache_key, es_search.previous_search_results, expire=timedelta(weeks=2))
+    variant_results = get_es_variants(
+        search_model.families.all(), parsed_search, user, previous_search_results, sort=sort, num_results=num_results,
+        **kwargs,
+    )
 
-    return variant_results, es_search.previous_search_results.get('total_results')
+    cache_key = _get_search_cache_key(search_model, sort=sort)
+    safe_redis_set_json(cache_key, previous_search_results, expire=timedelta(weeks=2))
+
+    return variant_results, previous_search_results.get('total_results')
 
 
 def get_variant_query_gene_counts(search_model, user):
-    gene_counts, _ = query_variants(search_model, es_search_cls=EsGeneAggSearch, sort=None, user=user)
+    previous_search_results = _get_cached_search_results(search_model)
+    if previous_search_results.get('gene_aggs'):
+        return previous_search_results['gene_aggs']
+
+    if len(previous_search_results.get('all_results', [])) == previous_search_results.get('total_results'):
+        return _get_gene_aggs_for_cached_variants(previous_search_results)
+
+    previously_loaded_results = process_es_previously_loaded_gene_aggs(previous_search_results)
+    if previously_loaded_results is not None:
+        return previously_loaded_results
+
+    gene_counts, _ = _query_variants(search_model, user, previous_search_results, gene_agg=True)
     return gene_counts
```

Comment:
The rew return values are not `counts`. The name of the function could be more straightforward.